subprojects/docs/src/docs/userguide/application_plugin.adoc

```diff
@@ -138,8 +138,12 @@ _Depends on_: `jar`, `startScripts`
 +
 Creates a full distribution TAR archive including runtime libraries and OS specific scripts.
 
+[[sec:application_extension]]
+== Application extension
+
+The application plugin adds an extension to the project, which you can use to configure its behaviour. See the link:{groovyDslPath}/org.gradle.api.plugins.JavaApplication.html[JavaApplication] DSL documentation.
 
 [[sec:application_convention_properties]]
-== Convention properties
+== Deprecated convention properties
 
-The application plugin adds some properties to the project, which you can use to configure its behaviour. See the link:{groovyDslPath}/org.gradle.api.Project.html[Project] class in the API documentation.
+The application plugin also adds some convention properties to the project, which you can use to configure its behaviour. These are deprecated and superseded by the extension described above. See the link:{groovyDslPath}/org.gradle.api.plugins.ApplicationPluginConvention.html[ApplicationPluginConvention] DSL documentation.
```

Comment:
Follow with something like:

> These properties appear as project properties in the build script. For example, to change the main class, you just write the following:
> ```
> mainClassName = "org.example.MyApp"
> ```
> **Note** You should configure the plugin via the extension described previously rather than these convention properties.

That last bit could even be added as a callout.